blog/2017-02-21-jest-19-immersive-watch-mode-test-platform-improvements.md

```diff
@@ -0,0 +1,159 @@
+
+## Immersive Watch Mode
+
+We [completely rewrote the watch mode](https://github.com/facebook/jest/pull/2362) to make it instant and more more extensible. As a result, the experience of using it really is immersive: tests re-run instantly after a file change and we added a typeahead to make it easy to select the right tests.
+
+### Filename pattern typeahead
+
+Filtering by filename now has a typeahead. This is particularly useful in large codebases to tell to exactly which files match your current pattern.
+
+![typeahead-pattern](/jest/img/blog/19-typeahead-pattern.gif)
+
+### Cached test name pattern typeahead.
+
+After running tests once, Jest knows which specific tests are available so we also added a typeahead for running specific tests within test suites:
+![typeahead-name](/jest/img/blog/19-typeahead-name.gif)
+
+<!--truncate-->
+
+## Snapshot Updates
+
+We made a couple of changes to the snapshot format. We don't make changes like this often and only consider them if they actually improve how snapshots work. As well as introducing a snapshot version number we accumulated a number of changes we wanted to make to the format for a while:
+
+* We dropped the “test” prefix in snapshot names from top level `test` or `it` calls.
+* We improved the printing of React elements to cause less changes when the last prop in an element changes.
+* We improved the character escaping mechanism to be more bulletproof.
+
+Before:
+
+```
+exports[`test snap 1`] = `
+<header>
+  <h1>
+    Jest \"19\"
+  </h1>
+  <Subtitle
+    name="Painless JavaScript Testing" />
+</header>
+`;
+```
+
+After (no “test” prefix, better JSX rendering, version header):
+
+```
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`snap 1`] = `
+<header>
+  <h1>
+    Jest "19"
+  </h1>
+  <Subtitle
+    name="Painless JavaScript Testing"
+  />
+</header>
+`;
+```
+
+We decided it's a good time to introduce versioned snapshots to ensure all developers are using a compatible version of Jest. Here's how we warn you about the need to update your snapshot:
+
+![snapshot-version](/jest/img/blog/19-snapshot-version.png)
+
+Please make sure you revert any local changes before updating to make the transition smooth and to ensure you aren't including any unwanted changes from failing tests in your new snapshots.
+
+## Improved printing of skipped tests
+
+Skipped tests are now printed as a single line instead of showing every individual one when testing in verbose mode or a single suite. Hopefully it will let you focus on currently important tests. It also occupies far less space!
+
+![skipped-tests](/jest/img/blog/19-skipped-tests.png)
+
+## New CLI arguments
+
+Jest 19 ships with two new coverage-related arguments which you can run from CLI:
+
+* `--collectCoverageFrom`
+* `--coverageDirectory`
+
+We now also error on invalid CLI arguments instead of ignoring them. But we've got your back with helpful error message like the one below, e.g. when you try running `jest --watc`:
```

Comment:
Also need to make "error message" plural (i.e. `s/message/messages/`), so it reads `But we've got your back with helpful error messages like the one below`